### ISSUE TITLE
Recursive mkdir() fails if the last dirname is '0'

### ISSUE THREAD

**Issue by andreasph** [mikey179/vfsStream#131]
Hello,
whenever I try to call a mkdir() for a folder structure with recursive option to true if the last subdirectory in the structure is equal to string '0' then the this final subdir is not created. It doesn't return any error but when I try to copy() a file to that subdir it fails with a message stating: 'failed to open stream: "org\bovigo\vfs\vfsStreamWrapper::stream_open" call failed'

I can locate the issue on vfsStream.php line 361, where a check is done for !empty($subDirs) (and since empty('0') returns true it skip the creation of the subdir)

````
public function testVfs()
    {
        org\bovigo\vfs\vfsStream::setup(
            'test',
            null,
            array(
                'tmp' => array(
                    'dummy.txt' => 'dummy content',
                ),
                'upload' => array(),
            )
        );

        $dummy = vfsStream::url('test/tmp/dummy.txt');
        $subdir = vfsStream::url('test/upload/a/b/0');
        $newfile = vfsStream::url('test/upload/a/b/0/newfile.xt');
        mkdir($subdir, 0777, true);
        copy($dummy, $newfile);
    }
````

**Comment by mikey179** [comment]
Thanks for the report! Fixed in release 1.6.3, available now.

**Comment by andreasph** [comment]
thanks for the quick fix
